cineast-core/src/main/java/org/vitrivr/cineast/core/iiif/IIIFConfig.java

```diff
@@ -0,0 +1,190 @@
+package org.vitrivr.cineast.core.iiif;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+import org.vitrivr.cineast.core.iiif.imageapi.ImageApiVersion;
+
+/**
+ * IIIF configuration used to fetch media files from remote servers.
+ *
+ * @author singaltanmay
+ * @version 1.0
+ * @created 28.05.21
+ */
+public class IIIFConfig {
+
+  @JsonProperty
+  private String imageApiVersion;
+  @JsonProperty(value = "url", required = true)
+  private String baseUrl;
+  @JsonProperty
+  private String region;
+  @JsonProperty
+  private String size;
+  @JsonProperty
+  private Float rotation;
+  @JsonProperty
+  private String quality;
+  @JsonProperty
+  private String format;
+  @JsonProperty("items")
+  private List<IIIFItem> iiifItems;
+
+  public ImageApiVersion getImageApiVersion() {
+    if (imageApiVersion == null) {
+      throw new IllegalArgumentException("Image API Version is not defined!");
+    }
+    return ImageApiVersion.fromNumericString(imageApiVersion);
+  }
+
+  @JsonProperty
+  public void setImageApiVersion(final String imageApiVersion) {
+    this.imageApiVersion = imageApiVersion;
+  }
+
+  @Override
+  public java.lang.String toString() {
+    return "IIIFConfig(imageApiVersion=" + this.getImageApiVersion() + ", baseUrl=" + this.getBaseUrl() + ", region=" + this.getRegion() + ", size=" + this.getSize() + ", rotation=" + this.getRotation() + ", quality=" + this.getQuality() + ", format=" + this.getFormat() + ", iiifItems=" + this.getIiifItems() + ")";
```

Comment:
Use ` return ReflectionToStringBuilder.toString(this, ToStringStyle.JSON_STYLE);` - this is something we're trying to standardize (see e.g. https://github.com/vitrivr/cineast/pull/204)

Reply:
Done